pkg/pillar/cmd/zedagent/zedagent.go

```diff
@@ -1273,7 +1280,7 @@ func triggerPublishDevInfo(ctxPtr *zedagentContext) {
 
 	log.Function("Triggered PublishDeviceInfo")
 	select {
-	case ctxPtr.TriggerDeviceInfo <- struct{}{}:
+	case ctxPtr.TriggerInfo <- infoForObjectKey{infoType: info.ZInfoTypes_ZiDevice}:
```

Comment:
Hmm, the assumption for this conditional send on the channel is that the channel is only used for device info messages, where we just need to make sure that the latest state is sent. Hence N calls to triggerPublishDevInfo can result in only two dev info messages being sent, but the second will be sent containing the state after the last call to triggerPublishDevInfo.

But now you are using the same channel for other messages hence this conditional send might fail to get the latest devinfo sent if the other sequence of triggers have been kicked off.

Easiest way would be to have two functions and channels - one for dev info (unchanged from master) plus the new channel and function you added to send the other messages.